src/transformers/models/regnet/modeling_tf_regnet.py

```diff
@@ -402,6 +298,12 @@ def call(
         )
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 
+        batch_size, num_channels, height, width = shape_list(pixel_values)
+        if tf.executing_eagerly() and num_channels != self.config.num_channels:
+            raise ValueError(
+                "Make sure that the channel dimension of the pixel values match with the one set in the configuration."
+            )
+
```

Comment:
I add this channel check in the stem (embedding) class, rather than here.

Could you also do that for consistency?

Also, `num_channels = shape_list(pixel_values)[1]` probably suffices.

Reply:
Done!